lib/linguist/repository.rb

```diff
@@ -132,7 +132,7 @@ def current_tree
     MAX_TREE_SIZE = 100_000
 
     def compute_stats(old_commit_oid, cache = nil)
-      return {} if current_tree.count_recursive(MAX_TREE_SIZE) >= MAX_TREE_SIZE
+      raise "Your repo tree size exceeds the MAX_LIMIT" if current_tree.count_recursive(MAX_TREE_SIZE) >= MAX_TREE_SIZE
```

Comment:
I'd reword this to be:

~~~ruby
raise "Repository tree size exceeds MAX_LIMIT (#{MAX_LIMIT})"
~~~

Also, did you mean `MAX_TREE_SIZE ` instead of `MAX_LIMIT`? I don't see the latter being used here.